tests/unit/test_languageHandler.py

```diff
@@ -31,3 +98,119 @@ def test_nonStandardLocale(self):
 	def test_invalidLocale(self):
 		lcid = languageHandler.localeNameToWindowsLCID("zzzz")
 		self.assertEqual(lcid, LCID_NONE)
+
+
+class TestSetLocale(unittest.TestCase):
+	"""
+	tests setting the python locale for all possible locales set by NVDA or the System
+	"""
+
+	SUPPORTED_LOCALES = [("en", "en_US"), ("fa-IR", "fa_IR"), ("an-ES", "an_ES")]
+
+	@classmethod
+	def tearDownClass(cls):
+		languageHandler.setLanguage(languageHandler.curLang)
+
+	def setUp(self):
+		languageHandler.setLanguage(languageHandler.curLang)
+
+	def testSupportedLocales(self):

```

Comment:
It would be good to have a little bit of an explanation on why the different variations of the tests exist.